oneview_redfish_toolkit/blueprints/resource_block.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (2018) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+
+from flask import abort
+from flask import Blueprint
+from flask import g
+from flask import Response
+from flask_api import status
+
+from hpOneView.exceptions import HPOneViewException
+from oneview_redfish_toolkit.api.errors import OneViewRedfishError
+from oneview_redfish_toolkit.api.server_hardware_resource_block \
+    import ServerHardwareResourceBlock
+
+
+resource_block = Blueprint("resource_block", __name__)
+
+
+@resource_block.route(
+    "/redfish/v1/CompositionService/ResourceBlocks/<uuid>",
+    methods=["GET"])
+def get_resource_block(uuid):
+    """Get the Redfish ResourceBlock for a given UUID.
+
+        Return ResourceBlock redfish JSON for a given UUID.
+        Logs exception of any error and return Internal Server
+        Error or Not Found.
+
+        Returns:
+            JSON: Redfish json with ResourceBlock.
+    """
+
+    try:
+        resource_index = g.oneview_client.index_resources.get_all(
+            filter='uuid=' + uuid
+        )
+
+        if resource_index:
+            category = resource_index[0]["category"]
```

Comment:
If the array is empty is already handled on line 96?

Reply:
The empty array case is handled by the if statement in the line above (line 53).